api4/user.go

```diff
@@ -1191,6 +1193,64 @@ func generateMfaSecret(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(secret.ToJson()))
 }
 
+func generateRecovery(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	if c.App.Session.IsOAuth {
+		c.SetPermissionError(model.PERMISSION_EDIT_OTHER_USERS)
+		c.Err.DetailedError += ", attempted access by oauth app"
+		return
+	}
+
+	if !c.App.SessionHasPermissionToUser(c.App.Session, c.Params.UserId) {
+		c.SetPermissionError(model.PERMISSION_EDIT_OTHER_USERS)
+		return
+	}
+
+	codes, err := c.App.GenerateRecovery(c.Params.UserId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+	jsonCodes, _ := json.Marshal(codes)
+
+	w.Header().Set("Cache-Control", "no-cache")
+	w.Header().Set("Pragma", "no-cache")
+	w.Header().Set("Expires", "0")
+	w.Write([]byte(jsonCodes))
+}
+
+func loginWithRecovery(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
I would propose that we don't add a extra handler, but instead allow users to pass the recovery token in the normal 2FA field. This way we don't need to add a extra handler and also save quite a bit of logic on the frontend.